portfolio/src/main/webapp/script.js

```diff
@@ -33,20 +33,26 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-// Fetch header for each page.
-fetch("templates/header.html")
-  .then(response => response.text())
-  .then(text => {
-    let header = document.createElement("header");
-    header.innerHTML = text;
-    document.querySelector("body").before(header);
-  });
-
-// Fetch footer for each page
-fetch("templates/footer.html")
-  .then(response => response.text())
-  .then(text => {
-    let footer = document.createElement("footer");
-    footer.innerHTML = text;
-    document.querySelector("body").after(footer);
-  });
\ No newline at end of file
+/**
+ * Loads the HTML retrieved from `path` and inserts it `where` relative to `selector`
+ * Example: loadHTML("templates/header.html", "body", "beforebegin") inserts the header before <body>
+ */
+function loadHTML(path, selector, where) {
```

Comment:
Supernit: don't quote me on this :) but I feel like parameter names are typically nouns?  Maybe something like "position" would be a better name, as the insertAdjacentHTML() method uses in its definition

Reply:
Good call. I was reconsidering this myself after work x) Made that change.